cmake/modules/SearchInstalledSoftware.cmake

```
@@ -642,7 +642,9 @@ if(mathmore OR builtin_gsl OR (tmva-cpu AND use_gsl_cblas))
     # FIXME: one need to find better way to extract path with GSL include files
     set(GSL_INCLUDE_DIR ${CMAKE_BINARY_DIR}/GSL-prefix/src/GSL-build)
     set(GSL_FOUND ON)
-    set(mathmore ON CACHE BOOL "Enabled because builtin_gls requested (${mathmore_description})" FORCE)
+    if(NOT mathmore)
+      message(WARNING "builtin_gls was enabled but mathmore wasn't")
+    endif()
```

Wouldn't that be 'better' or is `mathmore` already set on by default somewhere else?
```suggestion
    set(mathmore ON CACHE BOOL "Enabled because builtin_gls requested (${mathmore_description})")
    if(NOT mathmore)
      message(WARNING "builtin_gls was enabled but mathmore wasn't")
    endif()
```

---

we have `ROOT_BUILD_OPTION(mathmore OFF "Build libMathMore extended math library (requires GSL) [GPL]")`
unless -Dall=ON then we have set(mathmore_defvalue ON)

But: tmva-cpu AND use_gsl_cblas could require builtin_gsl also.
So, it's not 100% legit to turn on mathmore just because builtin_gsl is ON ?

This goes with the general philosophy of this PR. Builtins are hierarchically lower than "features". You turn on or off features, which may or not use builtins. Turning ON builtins cannot turn ON features automatically, you will just get a warning (or a cmake error if you prefer), whereas turning ON features can force-enable builtins without further thoughts.

---

> This goes with the general philosophy of this PR. 

Indeed, I agree with the first part (`disabling the feature should disable the feature`) but not with the second part (`enabling builtin without *also* enabling the feature fails to do what the user likely meant` [there is editorial comment in the 2nd description :) :) ] )